src/models/PrimaryAuth.js

```diff
@@ -106,12 +106,19 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
 
       var primaryAuthPromise;
 
-      if (this.appState.get('isUnauthenticated') && !this.settings.get('features.passwordlessAuth')) {
-        primaryAuthPromise = this.doTransaction(function (transaction) {
-          var authClient = this.appState.settings.authClient;
-          return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
-        });
-      } else {
+      if (this.appState.get('isUnauthenticated')) {
+        var authClient = this.appState.settings.authClient;
+        // bootstrapped with stateToken
+        if (!this.settings.get('features.passwordlessAuth')) {
+          primaryAuthPromise = this.doTransaction(function (transaction) {
```

Comment:
line 113-115 is identical to `doPrimaryAuthForPasswordless`. 
Did you miss some logic to pass `stateToken` or did I miss something here?

Reply:
I can clean this up. But basically when bootstrapped with a stateToken with or without passwordless auth we need to call `transaction.authenticate`, and auth JS will make sure we are posting the stateToken